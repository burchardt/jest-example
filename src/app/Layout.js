import React from "react";
import { MyButton } from './MyButton';
import { MyHeader } from './MyHeader';

export default class Layout extends React.Component {
	render() {
	    return(
	        <div>
	            <MyHeader />
	            <MyButton />
	        </div>
	    );
	}
}