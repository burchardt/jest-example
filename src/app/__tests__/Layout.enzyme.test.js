import React from 'react';
import { shallow, mount} from 'enzyme';

import Layout from '../Layout';

describe('<Layout />', () => {
    it('should render a shallow snapshot with no children', () => {
        const wrapper = shallow(
            <Layout />
        );
        // Generate and match a shallow snapshot
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a deep snapshot with its children', () => {
        const wrapper = mount(
            <Layout />
        );
        // Generate and match a deep snapshot
        expect(wrapper).toMatchSnapshot();
    });
});