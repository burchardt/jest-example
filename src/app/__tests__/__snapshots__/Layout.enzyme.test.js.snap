// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render a deep snapshot with its children 1`] = `
<Layout>
  <div>
    <MyHeader
      header="Header"
    >
      <h1>
        Header
      </h1>
    </MyHeader>
    <MyButton>
      <div>
        <button
          className="btn"
          id="myButton"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "lightblue",
              "fontSize": "25px",
            }
          }
        >
          Open
        </button>
      </div>
    </MyButton>
  </div>
</Layout>
`;

exports[`<Layout /> should render a shallow snapshot with no children 1`] = `
<div>
  <MyHeader
    header="Header"
  />
  <MyButton />
</div>
`;
